WalletWasabi/WabiSabi/Models/DecompositionAlgs/Coin.cs

```diff
@@ -0,0 +1,40 @@
+using NBitcoin;
+using System;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public record Coin : IComparable
+	{
+		public Money Amount { get; init; } = Money.Zero;
+		public const int OutputVbytes = 31;
+		public const decimal InputVbytes = 68.5m;
+
+		private int HashCode { get; } = new Guid().GetHashCode();
+
+		public Money EffectiveValue(FeeRate feeRate)
+		{
+			long sats = (long)Math.Floor(feeRate.SatoshiPerByte * InputVbytes);
+			return Amount - Money.Satoshis(sats);
+		}
+
+		public Money EffectiveCost(FeeRate feeRate)
+		{
+			long sats = feeRate.GetFee(OutputVbytes);
+			return Amount + Money.Satoshis(sats);
+		}
+
+		public override int GetHashCode()
+		{
+			return HashCode;
+		}
+
+		public int CompareTo(object? obj)
+		{
+			if (obj is Money money)
+			{
+				return CompareTo(money);
+			}
+			throw new NullReferenceException();
+		}
```

Comment:
I think this is not necessary.

Reply:
BinarySearch is using the IComparable interface. 

https://github.com/zkSNACKs/WalletWasabi/blob/6b49bee49a9aa6f73813c21a47aac88359c5ba7f/WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs#L15